### PR TITLE
Introduce `arizona.erl` and move (opaque) types inside it

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
# Description

We'll later add setters, getters and more interface definition to module `arizona`. This is just a minor first step to understand how consumption would happen next to `arizona_example`.

The TODO was turned into an issue on GitHub.

## Further considerations

When trying to use the new types in `arizona_example` I had to overcome a few obstacles. I'll look at the changed files and try to "review" on some of the decisions, if warranted.

- [x] I have performed a self-review of my changes
- [x] I have read and understood the [contributing guidelines](/arizona-framework/arizona/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -51,6 +51,7 @@ jobs:
 
       - name: Continuous Integration
         run: |
+          rebar3 xref
```
Comment (paulo-ferraz-oliveira) [inline .github/workflows/ci.yml:54]: Because we can...

File: include/arizona.hrl [hunk 2]
```diff
@@ -1,4 +1,3 @@
--define(ARIZONA_LIVEVIEW(Str), (begin
-    {ok, Tree} = arizona_live_view:parse_str(Str, Macros),
-    Tree
-end)).
+-define(ARIZONA_LIVEVIEW(Macros, Str), (
+    arizona_live_view:parse_str(Str, Macros)
```
Comment (paulo-ferraz-oliveira) [inline include/arizona.hrl:2]: Opacity less broken (and one step in the direction of "macroless", if possible).

File: src/arizona.erl
Comment (paulo-ferraz-oliveira) [inline src/arizona.erl:1]: For the time being there's not much more, but this'll grow in time (especially when we start to improve the interface scope-definition).

File: include/arizona.hrl (hunk 2 above)
Comment (williamthome) [reply include/arizona.hrl:2]: I'm all in for "macroless".

**Review by williamthome — APPROVED** [review]
Great change!

**Comment by williamthome** [comment]
Unresolved conversations are simply explanations, so I am giving myself the right to resolve them and move forward to merge https://github.com/arizona-framework/arizona_example/pull/9.

**Pull request merged**